presto-native-execution/presto_cpp/main/TaskManager.cpp

```diff
@@ -803,6 +810,16 @@ void TaskManager::maybeStartNextQueuedTask() {
     LOG(INFO) << "TASK QUEUE: Picking task to start from the queue: "
               << taskToStart->info.taskId;
     startTaskLocked(taskToStart);
+    // Make sure we call 'no more splits' we might have received before the task
+    // started.
+    auto execTask = taskToStart->task;
+    if (execTask != nullptr) {
+      for (const auto& planNodeId :
+           taskToStart->planNodesWithDelayedNoMoreSplits_) {
+        execTask->noMoreSplits(planNodeId);
```

Comment:
Would it be okay if we do it in the original place?

Reply:
Hmm, I don't understand the question. Do what and which original place?